src/fauxtoshop.cpp

```diff
@@ -1,517 +1,487 @@
+void getRandomPixelColor(
+        const Grid<int> &original, Grid<int> &scatterGrid, int row, int col, int scatterRadius);
+void applyScatterFilter(GBufferedImage &img);
+void applyEdgeDetectionFilter(GBufferedImage &img);
+void applyGreenScreenFilter(GBufferedImage &img);
+void compareImages(const GBufferedImage& img, GWindow &gw);
+void saveImage(GBufferedImage &img);
+int calculateRBGColourDifference(int pixel1, int pixel2);
+void validateLocationInput(int& row, int& col);
+
+int main() {
+    // 1. Greet the user
+    cout << "Welcome to Fauxtoshop!" << endl;
+    // Basic setup of Graphics Window
+    GWindow gw;
+    gw.setTitle("Fauxtoshop");
+    gw.setExitOnClose(true);
+    gw.setVisible(true);
+    // Enable the same random numbers on each run
+    fakeRandomNumberGenerator();
+    // Set up for the main image
+    GBufferedImage img;
+
+    // Ask the user to open an image file
+    while (nameImageFileToOpen(
+               img,
+               "Enter name of image file to open ('?' to open file selector or blank to quit): ")){
+        openImage(img, gw);
+        // User confirms which filter options to use: scatter, edge detection, 'green screen',
+        // compare image
+        int optionNum = selectFilterOptions();
+
+        // Complete filtering effect based on filter choice:
+        // 1.scatter, 2. edge detection, 3. 'green screen', 4. compare image
+        switch (optionNum){
+            case 1:
+                applyScatterFilter(img);
+                // Prompt the save to ask if they would like to save the new image
+                saveImage(img);
+                break;
+            case 2:
+                applyEdgeDetectionFilter(img);
+                // Prompt the save to ask if they would like to save the new image
+                saveImage(img);
+                break;
+            case 3:
+                applyGreenScreenFilter(img);
+                // Prompt the save to ask if they would like to save the new image
+                saveImage(img);
+                break;
+            case 4:
+                compareImages(img,gw);
+                break;
+            default:
+                cout << "Not a valid option" << endl;
+                break;
+        }
+        // Print a blank line after the programme is run once and repeat sequence from the beginning
+        cout << endl;
+        gw.clear();
+     }
+    return 0;
+}
+
+// Supporting functions
+/*
+ * Function: nameImageFileToOpen
+ * Usage: Prompts the user to open an image based on the file name. Adds the image to the window if
+ * the image is found. If the user enters blank then no image is opened.
+ * Params: GBufferedImage, GWindow
+ * -----------------------------------------------------------------
+ * Returns: True if an image was opened, false otherwise.
+*/
+bool nameImageFileToOpen(
+        GBufferedImage &img, const string& prompt, bool exitWhenUserEntersBlank){
+    while (true){
```

Comment:
In C++, prefer having a space between the ')' and '{'. Same for all other cases below. Tip: doing a cmd + f for "){" could help you locate all instances of this. :)